Documentation/design-docs/framework-version-resolution.md

```diff
@@ -232,38 +239,54 @@ In addition to the above any framework reference with a pre-release version will
+| `Minor`               | `minor`         | `false`                                    |
+| `LatestMinor`         | `minor`         | `true`                                     |
+| `Major`               | `major`         | `false`                                    |
+| `LatestMajor`         | `major`         | `true`                                     |
 
 ### Framework reference conflict resolution
 If there are two references to the same framework name, then the host needs to resolve the potential conflict. The rules are:
 * Take the higher `version`
 * Validate that the reference with the lower `version` allows roll-forward to the higher version. If not, fail.
-* Take the more restrictive `rollForward` setting
+* Take the more restrictive `version_range` from the two
+* If `roll_forward_to_highest_version` is true on one of the framework references, apply the `true` value to the merged framework reference as well.
 
 The check whether the roll-forward is allowed follows the rules described above in the list of available settings for `rollForward`.
 
-To select the more restrictive `rollForward` setting the values are ordered like this:
-1. `Disable`
-1. `LatestPatch`
-1. `Minor`
-1. `LatestMinor`
-1. `Major`
-1. `LatestMajor`
-
-Setting which is higher in this list (lower order number) is more restrictive than the one which is lower. `Disable` is the most restrictive (no roll-forward allowed). `LatestMajor` is the least restrictive (any other setting is compatible with it). `Minor` and `LatestMinor` are more restrictive then `Major` because they don't allow different major version.
-
 For example:
 In this example the two framework references are for the same framework name.
 
 | First framework reference | Second framework reference | Resolved framework reference |
 | ------------------------- | -------------------------- | ---------------------------- |
-| 2.1.0 Minor               | 2.2.0 Major                | 2.2.0 Minor                  |
-| 2.1.0 Minor               | 3.0.0 Minor                | failure                      |
-| 2.1.0 LatestMajor         | 3.0.0 Minor                | 3.0.0 Minor                  |
-| 2.1.0 LatestMajor         | 3.1.2 Disable              | 3.1.2 Disable                |
+| `2.1.0 minor`             | `2.2.0 major`              | `2.2.0 minor`                |
+| `2.1.0 minor`             | `3.0.0 minor`              | failure                      |
+| `2.1.0 major highest`     | `3.0.0 minor`              | `3.0.0 minor highest`        |
+| `2.1.0 major highest`     | `3.1.2 exact`              | `3.1.2 exact highest`        |
+
+To maintain backward compatibility, each framework reference also carries `applyPatches` setting. In case of two references the more restrictive setting value is used. So if one of the two framework references has `applyPatches=false` then the resolved framework reference also has `applyPatches=false`.
```

Comment:
How does this play with `LatestPatch` per the description above where `roll_to_highest_version` is ignored for `patch` versions?
e.g. for `version_range=patch`, `roll_to_highest_version=false`, `applyPatches=false`, based on the description about `version=patch` always picking latest patch regardless of `roll_to_highest_version`, it should pick the latest patch, but based on the `applyPatches` setting, it should not?

Reply:
I agree this is confusing - but hopefully not to end users as mixing `applyPatches` with `rollForward` should not really happen (we try to prevent it by not allowing the two settings to be specified in the same file).
The `applyPatches` "wins" over the `LatestPatch`. 

I added a description of this to the place which states that roll to latest patch is always used.